src/app/portlets/dot-templates/dot-template-advanced/dot-template-advanced.component.ts

```diff
@@ -0,0 +1,135 @@
+            }
+        ],
+        cancel: () => {
+            this.dotRouterService.goToURL('/c/templates');
+        }
+    };
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+    private originalValue: DotTemplate;
+
+    constructor(
+        private fb: FormBuilder,
+        private dotTemplateService: DotTemplatesService,
+        private dotRouterService: DotRouterService
+    ) {}
+
+    ngOnInit(): void {
+        this.group = this.fb.group({
+            title: ['', Validators.required],
+            friendlyName: '',
+            body: ['', Validators.required]
+        });
+
+        this.setFormOriginalValue();
+        this.onFormValueChanges();
+    }
+
+    ngOnDestroy(): void {
+        this.destroy$.next(true);
+        this.destroy$.complete();
+    }
+    /**
+     * This method initializes the monaco editor
+     *
+     * @param {*} editor
+     * @memberof DotTemplateComponent
+     */
+    initEditor(editor: any): void {
+        this.editor = editor;
+    }
+
+    /**
+     * This method creates the template
+     *
+     * @memberof DotTemplateComponent
+     */
+    saveTemplate(): void {
+        this.dotTemplateService
+            .create(this.group.value)
+            .pipe(take(1))
+            .subscribe(() => {
+                if (this.group.valid) {
+                    this.dotRouterService.goToURL('/c/templates');
+                }
+            });
+    }
+    /**
+     * This method handles the change event of the searchable selector and
+     * inserts the container to the editor
+     *
+     * @param {DotContainer} container
+     * @memberof DotTemplateComponent
+     */
+    containerChange(container: DotContainer): void {
+        const selection = this.editor.getSelection();
+
+        const id = this.setContainerId(container);
+
+        const text = `## Container: ${
+            container.name
+        }\n## This is autogenerated code that cannot be changed\n#parseContainer('${id}','${Date.now()}')\n`;
+        const operation = { range: selection, text: text, forceMoveMarkers: true };
+        this.editor.executeEdits('source', [operation]);
+    }
+
+    private setFormOriginalValue() {
+        this.originalValue = this.group.value;
+    }
+
+    private onFormValueChanges() {
+        this.group.valueChanges.pipe(takeUntil(this.destroy$)).subscribe((values) => {
+            this.setSaveActionState(values, this.originalValue);
+        });
+    }
+
+    private setContainerId(container: DotContainer): string {
+        const regex = new RegExp('//' + container.parentPermissionable.hostname);
+        return container.identifier.includes(container.parentPermissionable.hostname)
+            ? container.identifier.replace(regex, '')
+            : container.identifier;
+    }
+
+    private setSaveActionState(values: DotTemplate, oldValues: DotTemplate): void {
+        const primaryActionIndex = this.actions.primary.findIndex(
+            (action) => action.label === 'Save'
```

Comment:
look like the condition will break if we are in other language.

Reply:
langs are still pending